docs/sdk-json-proposal/SDK_API_DEFINITION.md

```diff
@@ -0,0 +1,251 @@
+# Mesh Data Plane SDK Definition
+
+## mesh_create_client()
+```c
+int mesh_create_client(MeshClient **client,
+                       const char *config_json)
+```
+Creates a new mesh client with the provided JSON configuration.
+
+#### Parameters
+* `[OUT]` `client` – Address of a pointer to a mesh client structure.
+* `[IN]` `config_json` – Pointer to a mesh client configuration string in JSON format.
+
+#### Client Configuration Example
+```json
+{
+  "apiVersion": "v1",
+  "apiConnectionString": "Server=192.168.95.1; Port=8001",
+  "apiDefaultTimeoutMicroseconds": 100000,
+  "maxMediaConnections": 32,
+}
+```
+
+#### Returns
+0 if successful. Otherwise, returns an error.
+
+
+## mesh_delete_client()
+```c
+int mesh_delete_client(MeshClient **client)
+```
+Deletes the mesh client and its resources.
+
+#### Parameters
+* `[IN/OUT]` `client` – Address of a pointer to a mesh client structure.
+
+#### Returns
+0 if successful. Otherwise, returns an error.
+
+
+## mesh_create_tx_connection()
+```c
+int mesh_create_tx_connection(MeshClient *client,
+                              MeshConnection **conn,
+                              const char *config_json)
+```
+Creates a transmitter connection with the provided JSON configuration.
+
+#### Parameters
+* `[IN]` `client` – Pointer to a parent mesh client.
+* `[OUT]` `conn` – Address of a pointer to the connection structure.
+* `[IN]` `config_json` – Pointer to a connection configuration structure.
+
+#### Transmitter/Receiver Connection Configuration Example
+```json
+{
+  "bufferQueueCapacity": 16,
+  "maxPayloadSize": 2097152,
+  "maxMetadataSize": 8192,
+  "connection": {
+    "multipoint-group": {
+      "urn": "ipv4:224.0.0.1:9003",
+    },
+    "st2110": {
+      "transport": "st2110-20" | "st2110-22" | "st2110-30" | "...",
+      "remoteIpAddr": "192.168.95.2",
+      "remotePort": "9002",
+      "pacing": "narrow",
+      "payloadType": 112
+    },
+    "rdma": {
+      "connectionMode": "RC" | "UC" | "UD" | "RD",
+      "maxLatencyNs": 10000
+    }
+  },
+  "payload": {
+    "video": {
+      "width": 1920,
+      "height": 1080,
+      "fps": 60.0,
```

Comment:
Shouldn't we use format like 60/10001, which is common in video encoding.
https://www.reddit.com/r/ffmpeg/comments/18j2yks/which_5994_is_the_real_5994/

Reply:
Eventually we can replace fps(float) to two integer based keys: fps_numerator(mandatory), fps_denominator(optional, default to 1). 
```
"fps_num": 60,
"fps_denom": 10001
```

Reply:
like it, but why 10001 instead of 1.0001?

Reply:
I agree that the only double fps value is better.